libs/cluster/Server/Replication/ReplicaOps/ReplicationReplicaAofSync.cs

```diff
@@ -48,14 +59,23 @@ public unsafe void ProcessPrimaryStream(byte* record, int recordLength, long pre
                             )
                         {
                             logger?.LogWarning("MainMemoryReplication: Skipping from {ReplicaReplicationOffset} to {currentAddress}", ReplicationOffset, currentAddress);
-                            storeWrapper.appendOnlyFile.Initialize(currentAddress, currentAddress);
+                            storeWrapper.appendOnlyFile.SafeInitialize(currentAddress, currentAddress);
                             ReplicationOffset = currentAddress;
                         }
                     }
                 }
 
-                // Address check
-                if (ReplicationOffset != storeWrapper.appendOnlyFile.TailAddress)
+                var tail = storeWrapper.appendOnlyFile.TailAddress;
+                var pageBits = storeWrapper.appendOnlyFile.UnsafeGetLogPageSizeBits();
+                var nextPageBeginAddress = ((tail >> pageBits) + 1) << pageBits;
+                if (tail + recordLength > nextPageBeginAddress && nextPageBeginAddress != currentAddress)
```

Comment:
if tail + recordLength <= nextPageBeginAddress, we need to still verify that tail == currentAddress

Reply:
i.e. we need to first check the common case tail == currentAddress. if that is true then all good. if not, then we need the overflow based check.

Reply:
I added a new check here https://github.com/microsoft/garnet/pull/783/commits/3fe5920d3dad3583cbe681cb129f76ccf6ad4e99.
Let me know if that looks good.